File: languages/python/bloock/entity/integrity/anchor_network.py
from __future__ import annotations

import bloock._bridge.proto.bloock_integrity_entities_pb2 as proto


class AnchorNetwork:
    """
    Represents information about an anchor network.
    """
    def __init__(self, name: str, state: str, tx_hash: str, root: str | None) -> None:
        """
        Constructs an AnchorNetwork object with the specified parameters.
        :type root: object
        :type tx_hash: object
        :type state: object
        :type name: object
        :rtype: object
        """
        self.name = name
        self.state = state
        self.tx_hash = tx_hash
        self.root = root

    @staticmethod
    def from_proto(network: proto.AnchorNetwork) -> AnchorNetwork:
        return AnchorNetwork(
            name=network.name, state=network.state, tx_hash=network.tx_hash, root=network.root
        )

    def to_proto(self) -> proto.AnchorNetwork:
        return proto.AnchorNetwork(
            name=self.name, state=self.state, tx_hash=self.tx_hash, root=self.root
        )

File: languages/php/Bloock/Entity/Integrity/AnchorNetwork.php
<?php

namespace Bloock\Entity\Integrity;

/**
 * Represents information about an anchor network.
 */
class AnchorNetwork
{
    private string $name;
    private string $state;
    private string $txHash;
    private ?string $root;

    /**
     * Constructs an AnchorNetwork object with the specified parameters.
     * @param string $name
     * @param string $state
     * @param string $txHash
     * @param string|null $root
     */
    public function __construct(string $name, string $state, string $txHash, ?string $root)
    {
        $this->name = $name;
        $this->state = $state;
        $this->txHash = $txHash;
        $this->root = $root;
    }

    public static function fromProto(\Bloock\AnchorNetwork $network): AnchorNetwork
    {
        return new AnchorNetwork($network->getName(), $network->getState(), $network->getTxHash(), $network->getRoot());
    }

    /**
     * Gets name of the anchor network.
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * Gets state of the anchor network.
     * @return string
     */
    public function getState(): string
    {
        return $this->state;
    }

    /**
     * Gets transaction hash of the anchor network.
     * @return string
     */
    public function getTxHash(): string
    {
        return $this->txHash;
    }

    /**
     * Gets root of the anchor network.
     * @return string
     */
    public function getRoot(): string
    {
        return $this->root;
    }

    public function toProto(): \Bloock\AnchorNetwork
    {
        $p = new \Bloock\AnchorNetwork();
        $p->setName($this->name);
        $p->setState($this->state);
        $p->setTxHash($this->txHash);
        $p->setRoot($this->root);
        return $p;
    }
}
